Source/drlg_l3.cpp

```diff
@@ -423,18 +423,19 @@ static void DRLG_L3River()
 			rx = 0;
 			ry = 0;
 			i = 0;
-			while ((dungeon[rx][ry] < 25 || dungeon[rx][ry] > 28) && i < 100) {
+			while ((dungeon[rx][ry] < 25 || dungeon[rx][ry] > 28) && i < 100) { /// BUGFIX: OOB `while (ry < DMAXY && (`

```

Comment:
Is the BUGFIX comment cut short? It ends with `&& (`

Reply:
It was meant to sign that it should just continue as previously
`while (ry < DMAXY && (dungeon[rx][ry] < 25 || dungeon[rx][ry] > 28) && i < 100) {v